docs(best-practices): conventional commit branch names (#20108)
renovatebot/renovate: docs/development/best-practices.md

> This document explains our best practices.

Author: I thought of putting the branch name stuff in the `local-development.md` and `remote-development.md` docs. But then I would have the same information in two files.

So I'm putting the branch name stuff in `best-practices.md` instead. 😄

Follow these best practices when you're working on our code.

> ## Git branch names

Author: I think it's a good idea to put this section near the top, as that way it's the first thing you see before you start creating branches. But if you want I can move it someplace else.

Branch names must start with a [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) scope like `feat/` or `fix/`.
If you're closing an issue with your PR then put the issue number after that.

For example:

- `feat/13732-cacache-cleanup`

> - `fix/gitea-automerge-strategy`

Reviewer: ```suggestion
- `fix/15431-gitea-automerge-strategy`
```

> You can help the maintainers by following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification when naming your Git branch.

Reviewer: Do you think we can remove this, and get to the point?

Author: ```suggestion
```

Yeah, let's remove the introduction and fast-forward to the instructions. I'm putting the link to the Conventional Commits spec to another sentence.

> Branch names should start with a Conventional Commit scope like `feat/` or `fix/` for example:

Author: ```suggestion
Branch names must start with a [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) scope like `feat/` or `fix/`.
If you're closing an issue with your PR: mention the GitHub issue number.

For example:
```

> - `feat/13732-cacache-cleanup`

Reviewer: Now I think about it, putting the issue number like above is important too, so we should explicitly mention it
